tt_metal/impl/dispatch/command_queue.cpp

```diff
@@ -981,13 +974,24 @@ void HWCommandQueue::enqueue_write_buffer(const Buffer& buffer, const void* src,
 
     void * final_src = (void *)src;
 
-    if (buffer.buffer_layout() == TensorMemoryLayout::WIDTH_SHARDED or
-        buffer.buffer_layout() == TensorMemoryLayout::BLOCK_SHARDED) {
-        final_src = convert_interleaved_to_sharded_on_host(src, buffer);
+    if (is_sharded(buffer.buffer_layout())) {
+        if (buffer.buffer_layout() == TensorMemoryLayout::WIDTH_SHARDED or
+            buffer.buffer_layout() == TensorMemoryLayout::BLOCK_SHARDED) {
+            final_src = malloc(buffer.num_dev_pages() * buffer.page_size());
+            convert_interleaved_to_sharded_on_host(final_src, src, buffer);
+        }
+        // HEIGHT SHARDED
+        // No swapping necessary but it adds padding
+        else  if (buffer.num_dev_pages() != buffer.num_pages()){
+            final_src = malloc(buffer.num_dev_pages() * buffer.page_size());
+            memcpy(final_src, src, buffer.num_pages() * buffer.page_size());
```

Comment:
This change + how it is read seems like it would cause worse perf for this height sharded case compared to before